src/components/RedditFooter/RedditFooter.tsx

```diff
@@ -1,84 +1,71 @@
+        <div className="link">
+          <button>
+            <FontAwesomeIcon icon="comment-alt"></FontAwesomeIcon>
+            <span className="link-text">{numString} comments</span>
+          </button>
+        </div>
 
-        if (this.state.saved) {
-            this.setState({saved: false});
-            this.redditHandler.unsave(listing).catch(() => {
-                this.setState({saved: true});
-            });
-        } else {
-            this.setState({saved: true});
-            this.redditHandler.save(listing).catch(() => {
-                this.setState({saved: false});
-            });
-        }
-    }
+        <div className="link">
+          <button>
+            <FontAwesomeIcon icon="award"></FontAwesomeIcon>
+            <span className="link-text">Give award</span>
+          </button>
+        </div>
 
-    renderFooter(numComments: number) {
-        let numString = util.normalizeNumber(numComments);
-        return (
-            <div className="footer">
-                <div className="link">
-                    <button>
-                        <FontAwesomeIcon icon="comment-alt"></FontAwesomeIcon>
-                        <span className="link-text">{numString} comments</span>
-                    </button>
-                </div>
+        <div className="link">
+          <button>
+            <FontAwesomeIcon icon="share"></FontAwesomeIcon>
+            <span className="link-text">Share</span>
+          </button>
+        </div>
 
-                <div className="link">
-                    <button>
-                        <FontAwesomeIcon icon="award"></FontAwesomeIcon>
-                        <span className="link-text">Give award</span>
-                    </button>
-                </div>
-                
-                <div className="link">
-                    <button>
-                        <FontAwesomeIcon icon="share"></FontAwesomeIcon>
-                        <span className="link-text">Share</span>
-                    </button>
-                </div>
+        <div className="link">
+          <button onClick={toggleSave}>
+            <FontAwesomeIcon
+              icon={saved ? "bookmark" : ["far", "bookmark"]}
+            ></FontAwesomeIcon>
+            <span className="link-text">{saved ? "Unsave" : "Save"}</span>
+          </button>
+        </div>
+      </div>
+    );
+  }
 
-                <div className="link">
-                    <button onClick={this.toggleSave}>
-                        <FontAwesomeIcon icon={this.state.saved ? 'bookmark' : ['far', 'bookmark']}></FontAwesomeIcon>
-                        <span className="link-text">{this.state.saved ? 'Unsave' : 'Save'}</span>
-                    </button>
-                </div>
-            </div>
-        )
-    }
-
-    render() {
-        const listing: Listing = this.props.listing;
-        return this.renderFooter(listing.num_comments);
-    }
+  return renderFooter(listing.num_comments);
```

Comment:
I don't think we need the renderFooter abstraction anymore. We should just get rid of that function and return that directly.